ide/editor.autosave/src/org/netbeans/modules/editor/autosave/AutoSaveController.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.editor.autosave;
+
+import java.beans.PropertyChangeListener;
+import java.io.IOException;
+import java.util.prefs.Preferences;
+import javax.swing.Timer;
+import javax.swing.text.Document;
+import javax.swing.text.JTextComponent;
+import org.netbeans.api.editor.EditorRegistry;
+import org.openide.LifecycleManager;
+import org.openide.cookies.SaveCookie;
+import org.openide.loaders.DataObject;
+import org.openide.modules.OnStart;
+import org.openide.modules.OnStop;
+import org.openide.util.Exceptions;
+import org.openide.util.NbPreferences;
+
+/**
+ *
+ * @author Michel Graciano
+ */
+public final class AutoSaveController {
+
+    public static final String KEY_ACTIVE = "autoSaveActive";
+    public static final String KEY_INTERVAL = "autoSaveInterval";
+    public static final String KEY_SAVE_ON_FOCUS_LOST = "autoSaveOnLostFocus";
+
+    public static final boolean KEY_ACTIVE_DEFAULT = true;
```

Comment:
Shall this be a constant? Anyway it is probably shall be false at first.
Remember people not using this module any more as they wished to use it. It is loaded by default.
We can enable autosave to be turned on  by default after circulating this function with the IDE for a while.Also probably autosave on focus lost should be enabled by default then instead of the timed save.

Reply:
Will we promote this new feature in the release? If so, I agree with you that it could be disabled by default, since users will know that there is this new feature. If not, maybe it is for the best to keep it enabled by default since it is like a feature to prevent people to lose unsaved work.

I have no objections with what you requested. I will disable both options by default and push it soon.

Reply:
Well, we won't do an ad campaign on this, however it would get a screenshot and a description on the features page.

Reply:
Disabled this feature by default in the latest push.